### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by rlouf**
I updated the SGMCMC example, which now runs in no time:
- Evaluation now happens on test set
- `jax.jit` the step function

I also updated the Aesara example; Aesara has a much simpler JAX transpilation pipeline since `aesara=2.8.5`.

Finally I changed the tensorflow requirement to `tensorflow-cpu`. I leave #303 as a good first issue.

### CODE REVIEW

**Comment by rlouf** [comment]
Notes on the tests that take the longest:
- The elliptical slice sampler is slow because the slice sampler works "too well": NUTS crawls for the same number of steps versus a few seconds.
- The Hierarchical BNN notebook could be much faster, especially the `get_predictions` function. I suspect this is where most of the time is spent, and not the sampling.
